src/components/base/BaseButton.vue

```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts">
+import { computed, defineProps } from "vue";
+
+const props = defineProps({
+  buttonType: {
+    type: String,
+    default: "primary",
+  },
+  buttonSize: {
+    type: String,
+    default: "reg",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+const emit = defineEmits(["click"]);
+
+const buttonClasses = computed(() => {
+  return [
+    "button",
+    { "button--primary": props.buttonType === "primary" },
+    { "button--primary-rev": props.buttonType === "primary-rev" },
+    { "button--secondary": props.buttonType === "secondary" },
+    { "button--tertiary": props.buttonType === "tertiary" },
+    { "button--tertiary-rev": props.buttonType === "tertiary-rev" },
+    { "button--sml": props.buttonSize === "sml" },
+    { "button--reg": props.buttonSize === "reg" },
+    { "button--med": props.buttonSize === "med" },
+    { "button--lrg": props.buttonSize === "lrg" },
+  ];
+});
+
+const buttonClick = (event: Event) => {
+  emit("click", event);
+};
```

Comment:
If this is just re-emiting of the event, we *could* leave it in the template - as in `@click="$emit('click', $event)"`